pkg/spireentry/reconciler.go

```diff
@@ -232,30 +234,34 @@ func (r *entryReconciler) getUnsupportedFields(ctx context.Context) (map[spireap
 	return r.config.EntryClient.GetUnsupportedFields(ctx, r.config.TrustDomain.Name())
 }
 
-func (r *entryReconciler) listClusterStaticEntries(ctx context.Context) ([]*ClusterStaticEntry, error) {
+func (r *entryReconciler) listClusterStaticEntries(ctx context.Context, className string, missingClassName bool) ([]*ClusterStaticEntry, error) {
 	clusterStaticEntries, err := k8sapi.ListClusterStaticEntries(ctx, r.config.K8sClient)
 	if err != nil {
 		return nil, err
 	}
 	out := make([]*ClusterStaticEntry, 0, len(clusterStaticEntries))
 	for _, clusterStaticEntry := range clusterStaticEntries {
-		out = append(out, &ClusterStaticEntry{
-			ClusterStaticEntry: clusterStaticEntry,
-		})
+		if (clusterStaticEntry.Spec.ClassName == "" && missingClassName) || clusterStaticEntry.Spec.ClassName == className {
```

Comment:
nit: this check could move into a helper on reconciler to avoid repetition of this logic.

e.g.:
```suggestion
		if r.reconcilesClass(clusterStaticEntry.Spec.ClassName) {
```

Reply:
Done